plugins/catalog-react/src/utils/filters.ts

```diff
@@ -26,16 +26,29 @@ import {
   EntityUserFilter,
   UserListFilter,
 } from '../filters';
+import { merge } from 'lodash';
 
-export function reduceCatalogFilters(
-  filters: EntityFilter[],
-): Record<string, string | symbol | (string | symbol)[]> {
+export interface CatalogFilters {
+  [x: string]:
+    | string
+    | Record<string, string | symbol | (string | symbol)[]>
+    | undefined;
+  filter: Record<string, string | symbol | (string | symbol)[]>;
+  fullTextFilter?: {
+    term: string;
+  };
+}
+
+export function reduceCatalogFilters(filters: EntityFilter[]): CatalogFilters {
   return filters.reduce((compoundFilter, filter) => {
-    return {
-      ...compoundFilter,
-      ...(filter.getCatalogFilters ? filter.getCatalogFilters() : {}),
-    };
-  }, {} as Record<string, string | symbol | (string | symbol)[]>);
+    const addedFilters = {};
+    if (filter.getBackendRequestParameter) {
+      Object.assign(addedFilters, filter.getBackendRequestParameter());
+    } else if (filter.getCatalogFilters) {
+      Object.assign(addedFilters, { filter: filter.getCatalogFilters() });
+    }
+    return merge(compoundFilter, addedFilters);
+  }, {} as CatalogFilters);
```

Comment:
this could become 

```suggestion
  return {
       ...compoundFilter,
      ...(filter.getCatalogFilters ? filter.getCatalogFilters() : {}),
      ...(isEntityTextFilter(filter) && filter.getFullTextFilters())
  };
```

Reply:
This would require changing the `getCatalogFilters` methods to return a nested `{filter: ...}` value, as well as adjusting `getBackendCatalogFilters`. I don't think that's a worthwhile API shift

Reply:
sorry my bad I didn't mean that.
My thinking was to leave the reduce as it was and calculate the `fullTextFilter` outside of the reduce.

```ts
interface CatalogFilters {
  filter: Record<string, string | symbol | (string | symbol)[]>;
  fullTextFilter?: {
    term: string;
  };
}

export function reduceCatalogFilters(filters: EntityFilter[]): CatalogFilters {
  const filter = filters.reduce<CatalogFilters['filter']>((compoundFilter, f) => {
    return {
      ...compoundFilter,
      ...f.getCatalogFilters?.(),
    };
  }, {});

  const fullTextFilter = filters.find(isEntityTextFilter)?.getFullTextFilters();
  return { filter, fullTextFilter };
}

function isEntityTextFilter(t: EntityFilter): t is EntityTextFilter {
  return !!(t as EntityTextFilter).getFullTextFilters;
}
```

as I mentioned in other comments a generic `getBackendRequestParameter` makes things a bit harder

Reply:
Makes sense, updated in [bc42ec3](https://github.com/backstage/backstage/pull/22131/commits/bc42ec30b5c8feb20d5c4789cc561509ae649cfb) 😁